### PR TITLE
[react-native-svg-animated-linear-gradient]: new definition

### PR DESCRIPTION
**Pull request by EndBug**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by EndBug** [comment]
Hey @martsie, I've opened a PR to add this package to DefinitelyTyped (I'm pinging you because you commented on https://github.com/FullstackStation/react-native-svg-animated-linear-gradient/pull/33)
It would be great if you could review the PR or, even better, join as the second owner of the definitions!

File: types/react-native-svg-animated-linear-gradient/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    primaryColor?: string;
+    secondaryColor?: string;
+    duration?: number;
+    width?: number | string;
+    height?: number | string;
+    x1?: string;
+    y1?: string;
+    x2?: string;
+    y2?: string;
+    useNativeDriver?: boolean;
```
Comment (peterblazejewicz) [inline types/react-native-svg-animated-linear-gradient/index.d.ts:18]:
```suggestion
    primaryColor?: string | undefined;
    secondaryColor?: string | undefined;
    duration?: number | undefined;
    width?: number | string | undefined;
    height?: number | string | undefined;
    x1?: string | undefined;
    y1?: string | undefined;
    x2?: string | undefined;
    y2?: string | undefined;
    useNativeDriver?: boolean | undefined;
```
[you can optionally consider to support usage within projects with option exact optional property types enabled](https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=&cad=rja&uact=8&ved=2ahUKEwjYtJ2n76r3AhVMpYsKHebVAE4QFnoECAcQAQ&url=https%3A%2F%2Fwww.typescriptlang.org%2Fplay%2F4-4%2Fnew-ts-features%2Fexact-optional-properties.ts.html&usg=AOvVaw1ruqglvzjmZ9g_CPsbZwbr)

Comment (EndBug) [reply types/react-native-svg-animated-linear-gradient/index.d.ts]: Seems like a good idea!

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@EndBug thx!

**Comment by EndBug** [comment]
Ready to merge

**Pull request merged**